shesmu-server-ui/src/help.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  UIElement,
+  buttonIcon,
+  subscribedState,
+  addElements,
+  br,
+} from "./html.js";
+import { locallyStored } from "./io.js";
+import { Publisher, pubSubModel } from "./util.js";
+export type Area = "action" | "alert" | "olive" | "pauses" | "simulator";
+/**
+ * Describes an individual help message that should appear.
+ */
+interface HelpTip {
+  /**
+   * A short (one line) description of the feature
+   */
+  summary: string;
+  /**
+   * A long description of the feature
+   */
+  description: string;
+  /**
+   * The hotspot (_i.e._, UI location) where this help tip should be displayed.
+   */
+  hotspot: string | null;
+  /**
+   * A general category for the kind of information in this tip so that everything in a category can be marked as read at once.
+   */
+  area: Area;
+}
+/**
+ * All the help for the whole UI. Each outer array represents a release version of Shesmu.
```

Comment:
So as versions are released, this file will expand? Can't old tips be versioned like the rest of the code?
Also, why don't the text of the tips live with the definition of the widget? From a dev perspective it seems weird to me to have to find this file to look up the name given to the tip I want to display

Reply:
Yes, this will expand. If a user skips ahead multiple versions, I kind of want them to get all the tips, so that's why I had it live with the code. I'm not really sure how this works in other system.

I agree that it's awkward for the tips to not live with the widget, but I want to have the dashboard-level tips include the tips of their components, unless you think that's not worthwhile.

Reply:
Okay, so only 'new' tips would appear in each outer array, not a whole copy of the entire tip set? 

If Feature A is released in version 2 and gets a tip, but then in version 3 Feature A is rewritten and the behaviour changes and gets another new tip, would a user skipping from version 1 to version 3 see the now-erroneous tip for version 2?

Reply:
Hmm. If we deleted the erroneous version 2 tip, then yes, but if it wasn't the last tip, that would probably break everything. Maybe the answer is to add a deprecated flag that can be switched on for older tips.

Reply:
That might be a good idea

Reply:
Done. You can replace the block with `null` and it should do the right thing.